p2p/src/peer_manager/peerdb/mod.rs

```diff
@@ -534,24 +553,48 @@ impl<S: PeerDbStorage> PeerDb<S> {
 
     /// Changes the address state to banned
     pub fn ban(&mut self, address: BannableAddress) {
-        let ban_till = (self.time_getter.get_time() + *self.p2p_config.ban_duration)
+        let ban_till = (self.time_getter.get_time() + *self.p2p_config.ban_config.ban_duration)
             .expect("Ban duration is expected to be valid");
 
         update_db(&self.storage, |tx| {
             tx.add_banned_address(&address, ban_till)
         })
-        .expect("adding banned address is expected to succeed (ban_peer)");
+        .expect("adding banned address is expected to succeed");
 
         self.banned_addresses.insert(address, ban_till);
     }
 
     pub fn unban(&mut self, address: &BannableAddress) {
         update_db(&self.storage, |tx| tx.del_banned_address(address))
-            .expect("adding banned address is expected to succeed (ban_peer)");
+            .expect("removing banned address is expected to succeed");
 
         self.banned_addresses.remove(address);
     }
 
+    /// Checks if the given address is discouraged
+    pub fn is_address_discouraged(&self, address: &BannableAddress) -> bool {
+        self.discouraged_addresses.contains_key(address)
+    }
+
+    /// Changes the address state to discouraged
+    pub fn discourage(&mut self, address: BannableAddress) {
+        // TODO: do we need to prolong the discouragement if the peer continues misbehaving?
```

Comment:
Let's decide whether to apply or drop this TODO.

Reply:
I think a refresh of the timeout is in order.

Also please create an RPC function to list discouraged peers.